tornjak-frontend/src/components/spiffe-helper.js

```diff
@@ -114,64 +114,189 @@ class SpiffeHelper extends Component {
 
-      for (let i=0; i < nodeEntries.length; i++) {
-          for (let j=0; j < lambdas.length; j++) {
-              lambdas[j](nodeEntries[i])
-          }
+      let agentSelectors = new Set(agent.selectors.map(formatSelectors));
+      let isAssocWithAgent = e => {
+        let entrySelectors = new Set(e.selectors.map(formatSelectors))
+        if (isSuperset(agentSelectors, entrySelectors)) {
+          agentEntriesDict[agentId].push(e);
+        }
+      };
+      lambdas.push(isAssocWithAgent);
+    }
+
+    for (let i = 0; i < nodeEntries.length; i++) {
+      for (let j = 0; j < lambdas.length; j++) {
+        lambdas[j](nodeEntries[i])
       }
+    }
 
-      return agentEntriesDict
+    return agentEntriesDict
   }
 
   // getCanonicalAgentSpiffeid takes in a agent entry, and returns the first 
   // agent ID that is found associated with it.
   getCanonicalAgentSpiffeid(entry, agents) {
-      let entrySelectors = new Set(entry.selectors.map(formatSelectors));
-      let isAssocWithEntry = a => {
-          let agentSelectors = new Set(a.selectors.map(formatSelectors));
-          return isSuperset(agentSelectors, entrySelectors);
+    let entrySelectors = new Set(entry.selectors.map(formatSelectors));
+    let isAssocWithEntry = a => {
+      let agentSelectors = new Set(a.selectors.map(formatSelectors));
+      return isSuperset(agentSelectors, entrySelectors);
+    }
+
+    let filteredAgents = agents.filter(isAssocWithEntry);
+    if (filteredAgents.length > 0) {
+      return this.getAgentSpiffeid(filteredAgents[0]);
+    }
+    return "";
+  }
+
+  // numberAgentEntries takes in spiffe id of an agent and entries list 
+  // returns number of entries for an agent
+  numberAgentEntries(spiffeid, globalEntries) {
+    if (typeof globalEntries !== 'undefined') {
+      var entriesList = globalEntries.filter(entry => {
+        return spiffeid === (this.getEntryParentid(entry))
+      })
+      return entriesList.length
+    } else {
+      return 0
+    }
+  }
+
+  // numberClusterEntries takes in an entry cluster metadata and list of entries
+  // returns number of entries in a cluster
+  numberClusterEntries(entry, globalEntries) {
+    var entriesPerAgent = entry.agentsList.map(currentAgent => {
+      return this.numberAgentEntries(currentAgent, globalEntries);
+    })
+    var sum = entriesPerAgent.reduce((acc, curVal) => {
+      return acc + curVal;
+    }, 0)
+    return sum
+  }
+
+  // cluster takes in an entry cluster metadata and list of entries
+  // returns cluster metadata info for dashboard table
+  cluster(entry, globalEntries) {
+    return {
+      id: entry.name,
+      name: entry.name,
+      created: entry.creationTime,
+      numNodes: entry.agentsList.length,
+      numEntries: this.numberClusterEntries(entry, globalEntries),
+    }
+  }
+
+  // numberEntries takes in spiffe id of an agent, avialble agents' spiffeids and list of entries
+  // returns cluster metadata info for dashboard table
```

Comment:
would be helpful to state that `agentEntriesDict` is the output of the function `SpiffeHelper.getAgentsEntries`